pkgs/development/python-modules/craft-parts/bash-path.patch

```diff
@@ -39,3 +28,56 @@ index b8d8f11..fce0e72 100644
                  check=True,
                  capture_output=True,
                  text=True,
+diff --git a/tests/unit/executor/test_step_handler.py b/tests/unit/executor/test_step_handler.py
+index 4e73c2b..b762fb8 100644
+--- a/tests/unit/executor/test_step_handler.py
++++ b/tests/unit/executor/test_step_handler.py
+@@ -209,9 +209,10 @@ class TestStepHandlerBuiltins:
+ 
+         assert get_mode(build_script_path) == 0o755
+         with open(build_script_path) as file:
++            import shutil
+             assert file.read() == dedent(
+                 f"""\
+-                #!/bin/bash
++                #!{shutil.which('bash')}
```

Comment:
Why not simply `"bash"`, so you would not need a patch.

Reply:
Because it still wouldn't be available at `/bin/bash`, right? And `/usr/bin/env` doesn't seem to be available in the build env, either.

Reply:
But it would be available in the path, no? In any case, I might be wrong, bare with me ;)

Reply:
Fair enough, I've added the packages and reduced this to just a single test that I don't think it makes sense to patch in order for it to work here.

Reply:
Oh I see your point -- yes, that's true, but I'd still need to patch either way to remove the explicit references to `/bin/bash` and `/usr/bin/sh` and the path lookup for java :(